kotsadm/pkg/store/s3pg/version_store.go

```diff
@@ -267,7 +272,114 @@ func (s S3PGStore) GetAppVersionArchive(appID string, sequence int64) (string, e
 	return tmpDir, nil
 }
 
-func (s S3PGStore) CreateAppVersion(appID string, currentSequence *int64, appName string, appIcon string, kotsKinds *kotsutil.KotsKinds) (int64, error) {
+func (s S3PGStore) CreateAppVersion(appID string, currentSequence *int64, appName string, appIcon string, kotsKinds *kotsutil.KotsKinds, filesInDir string, gitops gitopstypes.DownstreamGitOps, source string) (int64, error) {
+	db := persistence.MustGetPGSession()
+
+	tx, err := db.Begin()
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to begin")
+	}
+	defer tx.Rollback()
+
+	newSequence, err := s.createAppVersion(tx, appID, currentSequence, appName, appIcon, kotsKinds)
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to create app version")
+	}
+
+	if err := s.CreateAppVersionArchive(appID, int64(newSequence), filesInDir); err != nil {
+		return int64(0), errors.Wrap(err, "failed to create app version archive")
+	}
+
+	previousArchiveDir := ""
+	if currentSequence != nil {
+		// Get the previous archive, we need this to calculate the diff
+		previousDir, err := s.GetAppVersionArchive(appID, *currentSequence)
+		if err != nil {
+			return int64(0), errors.Wrap(err, "failed to get previous archive")
+		}
+
+		previousArchiveDir = previousDir
+	}
+
+	downstreams, err := s.ListDownstreamsForApp(appID)
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to list downstreams")
+	}
+
+	if err := secrets.ReplaceSecretsInPath(filesInDir); err != nil {
```

Comment:
can this _somehow_ get out of the store?  it feels better in the version package to me

Reply:
this can move out to the caller.  `filesInDir` is available before this function is called.